Swabbr.Core/Helpers/TimeZoneInfoHelper.cs

```diff
@@ -26,11 +28,51 @@ public static string MapTimeZoneToStringOrNull(TimeZoneInfo timeZoneInfo)
             }
 
             var offset = timeZoneInfo.BaseUtcOffset;
+            var absHours = Math.Abs(offset.Hours);
 
-            var hours = (offset.Hours >= 10) ? $"{offset.Hours}" : $"0{offset.Hours}";
-            var minutes = (offset.Minutes >= 10) ? $"{offset.Minutes}" : $"0{offset.Minutes}";
+            var hours = (absHours >= 10) 
+                ? $"{absHours}" 
+                : $"0{absHours}";
+            var minutes = (offset.Minutes >= 10) 
+                ? $"{offset.Minutes}" 
+                : $"0{offset.Minutes}";
 
-            return $"UTC+{hours}:{minutes}";
+            return (offset.Hours >= 0)
+                ? $"+{hours}:{minutes}"
+                : $"-{hours}:{minutes}";
+        }
+
+        /// <summary>
+        ///     Map a formatted timezone string according to ISO 8601 to a
+        ///     <see cref="TimeZoneInfo"/> object.
+        /// </summary>
+        /// <remarks>
+        ///     The expected format is +xx:xx or -xx:xx which is relative to 
+        ///     UTC. If <paramref name="s"/> is null or empty, null is returned.
+        /// </remarks>
+        /// <param name="s">Formatted string.</param>
+        /// <returns>Mapped <see cref="TimeZoneInfo"/> or null.</returns>
+        public static TimeZoneInfo MapStringToTimeZone(string s)
+        {
+            if (string.IsNullOrEmpty(s))
+            {
+                return null;
```

Comment:
Returning `null` is questionable in almost any situation. Either return a default value or throw.